extractor/src/main/java/com/example/extractor/CheckstyleExampleExtractor.java

```diff
@@ -63,18 +66,74 @@ private CheckstyleExampleExtractor() {
+            // New functionality: process single input file
+            final String inputFilePath = args[2];  // Changed from args[1]
+            final String outputFilePath = args[3]; // Changed from args[2]
+            processInputFile(Paths.get(inputFilePath), Paths.get(outputFilePath));
+        } else {
+            // Functionality: process all examples
+            final String checkstyleRepoPath = args[0];
+            final List<Path> allExampleDirs = findAllExampleDirs(checkstyleRepoPath);
+            final Map<String, List<Path>> moduleExamples = processExampleDirs(allExampleDirs);
+
+            YamlParserAndProjectHandler.processProjectsForExamples(PROJECT_ROOT.toString());
+
+            for (final Map.Entry<String, List<Path>> entry : moduleExamples.entrySet()) {
+                generateAllInOneConfig(entry.getKey(), entry.getValue());
+                generateReadmes(entry.getKey(), entry.getValue());
+            }
+        }
+    }
 
-        YamlParserAndProjectHandler.processProjectsForExamples(PROJECT_ROOT.toString());
+    /**
+     * Processes an input file and generates an output file.
+     *
+     * @param inputFile  The path to the input file
+     * @param outputFile The path to the output file
+     * @throws Exception If an error occurs during processing
+     */
+    public static void processInputFile(final Path inputFile, final Path outputFile) throws Exception {
+        // Check if the input file exists
+        if (!Files.exists(inputFile)) {
+            LOGGER.severe("Input file does not exist: " + inputFile);
+            throw new IOException("Input file does not exist: " + inputFile);
+        }
+
+        // Get the template file path and serialize the configuration
+        final String templateFilePath = getTemplateFilePathForInputFile(inputFile.toString());
+        final String generatedContent = ConfigSerializer.serializeNonXmlConfigToString(inputFile.toString(), templateFilePath);
+
+        // Write the generated content to the output file
+        Files.writeString(outputFile, generatedContent, StandardCharsets.UTF_8);
+
+        LOGGER.info("Generated configuration at " + outputFile);
+    }
 
-        for (final Map.Entry<String, List<Path>> entry : moduleExamples.entrySet()) {
-            generateAllInOneConfig(entry.getKey(), entry.getValue());
-            generateReadmes(entry.getKey(), entry.getValue());
+    /**
+     * Retrieves the template file path based on the input file path.
+     *
+     * @param inputFilePath The path to the input file
+     * @return The template file path
+     */
+    public static String getTemplateFilePathForInputFile(final String inputFilePath) {
+        try {
+            final TestInputConfiguration testInputConfiguration = InlineConfigParser.parse(inputFilePath);
+            final List<ModuleInputConfiguration> modules = testInputConfiguration.getChildrenModules();
+
+            final ModuleInputConfiguration mainModule = modules.get(0);
```

Comment:
after initial implementation we should support multiple modules - https://github.com/checkstyle/test-configs/issues/157

Reply:
okay